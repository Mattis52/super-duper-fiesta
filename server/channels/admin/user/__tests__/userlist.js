jest.mock('../../../../models/meeting');
jest.mock('../../../../models/user');
jest.mock('../../../../utils');
const { requestUserList } = require('../userlist');
const { emit, broadcast } = require('../../../../utils');
const { getActiveGenfors } = require('../../../../models/meeting');
const { getUsers } = require('../../../../models/user');
const { generateUser, generateGenfors, generateSocket } = require('../../../../utils/generateTestData');

describe('requestUserList', () => {
  beforeEach(() => {
    getActiveGenfors.mockImplementation(async () => generateGenfors());
    getUsers.mockImplementation(async genfors => [
      generateUser({ genfors: genfors.id, _id: '1' }),
      generateUser({ genfors: genfors.id, _id: '2' }),
    ]);
  });

  const generateData = () => ({});

  it('emits user list action on success', async () => {
    await requestUserList(generateSocket(), generateData());

    expect(emit.mock.calls).toMatchSnapshot();
    expect(broadcast.mock.calls).toEqual([]);
  });

  it('emits error when no active genfors was found', async () => {
    getActiveGenfors.mockImplementation(async () => { throw new Error('Failed'); });
    await requestUserList(generateSocket(), generateData());

    expect(emit.mock.calls).toMatchSnapshot();
    expect(broadcast.mock.calls).toEqual([]);
  });

  it('emits error when retrieving users fails', async () => {
    getUsers.mockImplementation(async () => { throw new Error('Failed'); });
    await requestUserList(generateSocket(), generateData());

    expect(emit.mock.calls).toMatchSnapshot();
    expect(broadcast.mock.calls).toEqual([]);
  });
});
