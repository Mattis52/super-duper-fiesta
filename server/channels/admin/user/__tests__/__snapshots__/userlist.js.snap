// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requestUserList emits error when no active genfors was found 1`] = `Array []`;

exports[`requestUserList emits error when retrieving users fails 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "USER_LIST",
    Array [],
    Object {
      "error": "Could not fetch user list.",
    },
  ],
]
`;

exports[`requestUserList emits user list action on success 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "USER_LIST",
    Array [
      Object {
        "_id": "1",
        "completedRegistration": true,
        "genfors": "1",
        "logged_in": true,
        "name": "Namy",
        "onlinewebId": "123",
        "permissions": 5,
      },
      Object {
        "_id": "2",
        "completedRegistration": true,
        "genfors": "1",
        "logged_in": true,
        "name": "Namy",
        "onlinewebId": "123",
        "permissions": 5,
      },
    ],
  ],
]
`;
